sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledHashJoinExec.scala

```diff
@@ -317,4 +318,41 @@ case class ShuffledHashJoinExec(
       v => s"$v = $thisPlan.buildHashedRelation(inputs[1]);", forceInline = true)
     HashedRelationInfo(relationTerm, keyIsUnique = false, isEmpty = false)
   }
+
+  /**
+   * Generates the code for anti join.
+   * Handles NULL-aware anti join (NAAJ) separately here.
+   */
+  protected override def codegenAnti(ctx: CodegenContext, input: Seq[ExprCode]): String = {
+    if (isNullAwareAntiJoin) {
+      val HashedRelationInfo(relationTerm, _, _) = prepareRelation(ctx)
+      val (keyEv, anyNull) = genStreamSideJoinKey(ctx, input)
+      val (matched, _, _) = getJoinCondition(ctx, input)
+      val numOutput = metricTerm(ctx, "numOutputRows")
+      val found = ctx.freshName("found")
+
+      // Skip code generation for EmtpyHashedRelation and EmptyHashedRelationWithAllNullKeys cases,
+      // since it is already handled by AQE rewrite.
+      s"""
+         |boolean $found = false;
+         |// generate join key for stream side
+         |${keyEv.code}
+         |if ($anyNull) {
+         |  $found = true;
+         |} else {
+         |  UnsafeRow $matched = (UnsafeRow)$relationTerm.getValue(${keyEv.value});
```

Comment:
`getValue()` is for retrieving of unique key only. how about changing this into `get()` to be consistent with existing non-codegen code path. Though there's no issue to retrieve a single value for key, when key is not unique. But it can become a bug later when people enforce some check inside `HashedRelation`.